MinecraftClient/ChatBot.cs

```diff
@@ -984,12 +984,23 @@ protected Mapping.Location GetCurrentLocation()
         /// <param name="location">Location to reach</param>
         /// <param name="allowUnsafe">Allow possible but unsafe locations thay may hurt the player: lava, cactus...</param>
         /// <param name="allowDirectTeleport">Allow non-vanilla teleport instead of computing path, but may cause invalid moves and/or trigger anti-cheat plugins</param>
+        /// <param name="allowApproachIfGoalNotReached">If no valid path can be found return a path that approaches your goal</param>
```

Comment:
How about replacing the boolean with a parameter that states maximum offset instead?
```suggestion
        /// <param name="maxOffset">If no valid path can be found, also allow locations within specified distance of destination</param>
```
That should be helpful in the case discussed before where we want to walk to a destination that is far away and chunks are not loaded yet. This way it's possible to say e.g. `3` for going near some block to break, `Int.MaxValue` for calculating path until edge of chunk.